### PR TITLE
chore: release-it and semantic pr title

### PR DESCRIPTION
**Pull request by robertsLando**
This PR adds release-it module to handle releases and semantic PR title checks to improve changelogs.

Mind that in order to make the `release-it` action to work `NPM_TOKEN` secret must be created

Anyway a script `release` has been added to `package.json` that can be used to create releases interactively (much better then doing them manually like we do now)

In order to prevent overriding changelog, the old changelog has been renamed to `CHANGELOG.old.md`

I also did an `npm audit fix` that fixed 6 high security vulnerabilities

### CODE REVIEW

File: .github/workflows/release-it.yml
Comment (vishnureddy17) [inline .github/workflows/release-it.yml:16]: Is there a way to use `release-it` without hardcoding a GitHub username?

**Comment by vishnureddy17** [comment]
Could you write a brief markdown file describing how to release? I'm approving the PR anyway since my concerns are minor.

**Review by vishnureddy17 — DISMISSED** [review]

File: .github/workflows/release-it.yml
Comment (robertsLando) [reply .github/workflows/release-it.yml:16]: Yeah this is just to prevent someone else to be able to trigger releases, I can add your username if you want or completely remove this check

**Comment by robertsLando** [comment]
Ok for the markdown file, anyway releasing is as easy as typing "npm run release" and follow the interactive cli: it will firstly ask you the gh token then will do the tests, update changelog, create a release commit with release tag and create the release on gh and npm (will ask for the otp too).

File: .github/workflows/release-it.yml
Comment (vishnureddy17) [reply .github/workflows/release-it.yml:16]: They won't be able to do anything without write access to this repo or the npm package right? If that's the case, I vote to remove it.

**Comment by vishnureddy17** [comment]
> What about eslint+prettier in place of standard?

I think it should be a different PR, because I imagine changing linter and formatter will change a lot of the code, and I think it's better to have all those changes in its own commit

**Comment by robertsLando** [comment]
Agree!

Would you mind to add the 2 secrets mentioned above?

**Comment by vishnureddy17** [comment]
> Would you mind to add the 2 secrets mentioned above?

I've actually never used GitHub actions before so I will have to do some digging to understand how to make that work.

Is this PR blocked on that?

File: .github/workflows/release-it.yml
Comment (robertsLando) [reply .github/workflows/release-it.yml:16]: Yeah, ok!

**Comment by vishnureddy17** [comment]
@robertsLando Thanks! Gonna check it out anway since I should probably know this.

**Pull request merged**
